src/tests/JIT/HardwareIntrinsics/X86/X86Base/CpuId.cs

```diff
@@ -0,0 +1,176 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics.X86;
+using System.Runtime.Intrinsics;
+
+namespace IntelHardwareIntrinsicTest
+{
+    class Program
+    {
+        const int Pass = 100;
+        const int Fail = 0;
+
+        static unsafe int Main(string[] args)
+        {
+            int testResult = Pass;
+
+            if (!X86Base.IsSupported)
+            {
+                return testResult;
+            }
+
+            (int eax, int ebx, int ecx, int edx) = X86Base.CpuId(0x00000000, 0x00000000);
+
+            bool isAuthenticAmd = (ebx == 0x68747541) && (ecx == 0x444D4163) && (edx == 0x69746E65);
+            bool isGenuineIntel = (ebx == 0x756E6547) && (ecx == 0x6C65746E) && (edx == 0x6C656E69);
+
+            if (!isAuthenticAmd && !isGenuineIntel)
+            {
+                Console.WriteLine("Unrecognized CPU vendor");
```

Comment:
Would we hit this with e.g. Parallels or Hyper-V? I didn't check it, but Wikipedia says they report their own CPUIDs. Maybe checking for an alphanumeric ASCII range would be more reliable?

Reply:
I don't believe so, afaik Hyper-V reports the underlying CPU vendor (namely because it uses hardware accelerated virtualization). I don't know about Parallels.
* On both my AMD and Intel machines, Hyper-V is reporting the same underlying vendor name and model number/etc

Reply:
what about x86 emulation on ARM like on new Macs and Surface Pro X?

Reply:
If someone ends up running against a new CPU vendor that isn't AMD or Intel they would need to update the test and validate that the CPUID bits are correct/the same.
There is no guarantee, even between AMD and Intel, that a given CPUID call or bit is exactly identical (and in some cases the same bits do in fact represent different data).

Reply:
I added a comment to the test elaborating on this.